coordinator/persistence/aggregation/src/integrationTest/kotlin/net/consensys/zkevm/persistence/dao/aggregation/AggregationsPostgresDaoTest.kt

```diff
@@ -293,6 +293,90 @@ class AggregationsPostgresDaoTest : CleanDbTestSuiteParallel() {
     }
   }
 
+  @Test
+  fun findConsecutiveProvenBlobsReturnsFullListOfProvenBlobsWithProvenBatches() {
+    val batch1 = createBatch(0L, 10L)
+    val batch2 = createBatch(11L, 20L)
+    val batch3 = createBatch(21L, 30L)
+
+    val blob1 = createBlobRecordFromBatches(listOf(batch1))
+    val blob2 = createBlobRecordFromBatches(listOf(batch2, batch3))
+
+    val batches = listOf(batch1, batch2, batch3)
+    val blobs = listOf(blob1, blob2)
+
+    batches.forEach { insertBatch(it).get() }
+    blobs.forEach { insertBlob(it).get() }
+
+    aggregationsPostgresDaoImpl.findConsecutiveProvenBlobs(0).get().also { blobCounters ->
+      assertThat(blobCounters).hasSameElementsAs(
+        listOf(
+          getBlobAndBatchCounters(listOf(batch1), blob1),
+          getBlobAndBatchCounters(listOf(batch2, batch3), blob2)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun findConsecutiveProvenBlobsWhenDbIsEmpty() {
+    aggregationsPostgresDaoImpl.findConsecutiveProvenBlobs(0).get().also { blobCounters ->
+      assertThat(blobCounters).hasSameElementsAs(emptyList())
+    }
+  }
+
+  @Test
+  fun findConsecutiveProvenBlobsUnprovenBlobAndBatchAtDifferentIntervals() {
+    // Batch2 proof is missing
+    // Blob for batch3 is missing
+    val batch1 = createBatch(0L, 10L)
+    val batch2 = createBatch(11L, 20L, status = Batch.Status.Finalized)
+    val batch3 = createBatch(21L, 30L)
+    val batch4 = createBatch(31L, 40L)
+
+    val blob1 = createBlobRecordFromBatches(listOf(batch1, batch2))
+    val blob2 = createBlobRecordFromBatches(listOf(batch3), status = BlobStatus.COMPRESSION_PROVING)
+    val blob3 = createBlobRecordFromBatches(listOf(batch4))
+
+    val batches = listOf(batch1, batch2, batch3, batch4)
+    val blobs = listOf(blob1, blob2, blob3)
+
+    batches.forEach { insertBatch(it).get() }
+    blobs.forEach { insertBlob(it).get() }
+
+    aggregationsPostgresDaoImpl.findConsecutiveProvenBlobs(0).get().also { blobCounters ->
+      assertThat(blobCounters).hasSameElementsAs(emptyList())
```

Comment:
I still don't understand well what this test is trying to ensure. Can you please explain?

`findConsecutiveProvenBlobs(startBlockNumber)` is supposed to return result if there is a blob starting at `startBlockNumber` right? If so, shall we call it with 1 instead and return blob1 [0..20]?

Reply:
I think this test case is to test if the `findConsecutiveProvenBlobs()` would return the result correctly when some batches or blobs in the middle are unproven, even though unproven records will not exist in real scenario, but the dao/db was supporting them at the beginning, so this test is needed to cover that.

Reply:
For simplicity, I suggest to just remove the support in the DAO for unproven stuff:

- 1. we took the decision that we won't have unproven elements in the DB;
- 2. it's unnecessary complexity that we keep all over :(